api/cluster.go

```diff
@@ -86,8 +87,10 @@ func (s *Server) indexList(ctx *middleware.Context, req models.IndexList) {
 func (s *Server) getData(ctx *middleware.Context, request models.GetData) {
 	series, err := s.getTargetsLocal(request.Requests)
 	if err != nil {
+		// the only errors returned are from us catching panics, so we should treat them
+		// all as internalServerErrors
```

Comment:
I find this comment a bit misleading. they are all internal server errors because it's an internal endpoint. there can simply be no othere class of errors (e.g. no validation errors because we (rightfully) assume pre-validated requests). the fact that they're from recovering from panics has in my mind nothing to do with it, and we may (and mostly likely will) change that approach to return some errors directly (and they will still be internal server errors).  anyway this is only a minor point.